EJ 2

TheOneWhoBurns opened the pull request.

TheOneWhoBurns committed:
EJ 2

EstAlvB commented on EJ 2/EJ2.ipynb:
```diff
@@ -0,0 +1,33 @@
+ },
+ "nbformat": 4,
+ "nbformat_minor": 5
+}
```
No sé si solo yo veo así, pero parece que hubo un error y no se subió el código del ejercicio 2. ¿Por favor podrías revisarle si esto es así?

TheOneWhoBurns committed:
EJ2 Correguido + explicacion

aparentemente subi el EJ2 en el formato de diccionario y uno vacio, aqui esta en .py

TheOneWhoBurns committed:
Create ExplicacionEJ2.md

EstAlvB commented on EJ2/EJ2.py:
```diff
@@ -0,0 +1,51 @@
+class RandomizedSet:
+
+    def __init__(self):
+        self.index_map = {} #this one has {[index, value]}
```
Buena idea al usar diccionarios. El código se vuelve mucho más eficiente.

Spidyqda commented:
Me parece interesante como logras definir la función remove() ya que por lo que veo, eliminas con el método pop el valor deseado, especificando antes con el valor dado por el usuario y para sí encontrar su posición y así eliminarlo. De igual manera me gusta como logras actualizar el mapa.

TheOneWhoBurns commented:
Si, lo unico que hay que tener en cuenta es que no se desicnonizen los hash maps.
